src/inc/til/env.h

```diff
@@ -0,0 +1,515 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <wil/token_helpers.h>
+#include <winternl.h>
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    namespace details
+    {
+        //
+        // A case-insensitive wide-character map is used to store environment variables
+        // due to documented requirements:
+        //
+        //      "All strings in the environment block must be sorted alphabetically by name.
+        //      The sort is case-insensitive, Unicode order, without regard to locale.
+        //      Because the equal sign is a separator, it must not be used in the name of
+        //      an environment variable."
+        //      https://docs.microsoft.com/en-us/windows/desktop/ProcThread/changing-environment-variables
+        //
+        struct wstring_case_insensitive_compare
+        {
+            [[nodiscard]] bool operator()(const std::wstring& lhs, const std::wstring& rhs) const noexcept
+            {
+                return (::_wcsicmp(lhs.c_str(), rhs.c_str()) < 0);

```

Comment:
We may need to switch to CompareStringOrdinal to avoid involving locales; the C runtime library functions are evil

(I recognize I may be commenting on Michael-original code, and I appreciate you taking this PR and running with it 😄)

Reply:
`CompareStringOrdinal` can theoretically return an error result (0) if the parameters are incorrect (with the error obtainable using `GetLastError`) - what do you want to do in this situation?

The docs would appear to suggest this would likely only happen if you pass something that isn't `0` or `1` in the `bIgnoreCase` variable, but obviously there's no guarantee there.

https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-comparestringordinal#return-value

Reply:
I've wrapped it in a helper function and used `FAIL_FAST_LAST_ERROR_IF()`.